src/awkward/operations/ak_all.py

```diff
@@ -57,60 +56,35 @@ def all(
             axis=axis,
             keepdims=keepdims,
             mask_identity=mask_identity,
-            flatten_records=flatten_records,
             highlevel=highlevel,
             behavior=behavior,
         ),
     ):
-        return _impl(
-            array, axis, keepdims, mask_identity, flatten_records, highlevel, behavior
-        )
+        if flatten_records is not unset:
+            raise ak._errors.wrap_error(
+                ValueError(
+                    "`flatten_records` is no longer a supported argument for reducers. "
+                    "Instead, use `ak.ravel(array)` first to remove the record structure "
+                    "and flatten the array."
+                )
+            )
```

Comment:
We are hereby committing to a hard-deprecation. The function argument becomes unusable immediately. I think this is okay, considering how obscure the function argument is and how recently 2.0 was released.